File: array_and_hashtable/convert_an_array_into_2d_array_with_conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        HashMap<Integer, Integer> noOfOccurrences = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            int value = noOfOccurrences.getOrDefault(nums[i], 0) + 1;
            noOfOccurrences.put(nums[i], value);
        }
        System.out.println(noOfOccurrences);

        List<List<Integer>> res = new ArrayList<>();
        int maxOccurrences = noOfOccurrences.values().stream().max(Integer::compare).orElse(0);

        for (int i = 0; i < maxOccurrences; i++) {
            List<Integer> temp = new ArrayList<>();
            for (Map.Entry<Integer, Integer> entry : new ArrayList<>(noOfOccurrences.entrySet())) {
                int key = entry.getKey();
                int value = entry.getValue();
                if (value != 0) {
                    temp.add(key);
                }
                noOfOccurrences.put(key, value - 1);
                if (noOfOccurrences.get(key) == 0) {
                    noOfOccurrences.remove(key);
                }
            }
            res.add(temp);
        }

        return res;
    }
}

File: array_and_hashtable/convert_an_array_into_2d_array_with_conditions.py
from typing import List

class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        no_of_occurrences = {}
        for i in nums:
            no_of_occurrences[i] = no_of_occurrences.get(i,0) + 1
        
        res = []
        for _ in range(max(no_of_occurrences.values())):
            temp = []
            for key, value in list(no_of_occurrences.items()):
                if value != 0:
                    temp.append(key)
                no_of_occurrences[key] -= 1
                if no_of_occurrences[key] == 0:
                    del no_of_occurrences[key]
            res.append(temp)

        return res

